src/helpers/redux.js

```diff
@@ -0,0 +1,49 @@
+import dotProp from 'dot-prop-immutable';
+
+/**
+ * Creates a function which transforms arg list to objekt with followed keys
+ *
+ * Example:
+ *
+ * const changeParentPayloadCreator = withPayload('nodeId', 'nextParentId');
+ * const payload = changeParentPayloadCreator(1001, 2000);
+ * // payload = {
+ * //   nodeId: 1001,
+ * //   nextParentId: 2000,
+ * // }
+ *
+ * You can pass also an object in array if you want to pass some values to payload
+ */
+export const withPayload = (...keys) => {
+  const createPayload = (...args) => {
+    return keys.reduce((payload, property, index) => {
+      if (typeof property === 'object') {
+        Object.assign(payload, property);
+      } else {
+        payload[property] = args[index];
+      }
+
+      return payload;
+    }, {});
+  };
+
+  return createPayload;
+};
+
+/**
+ * Shorthand to update substate
+ *
+ * @param {*} name - property name to update
+ * @param {*|Function} updater - next value or function to update
+ */
+export const updateState = (name, updater) => {
+  const reducer = (state, { payload }) => {
+    const nextValue =
+      typeof updater === 'function' ? updater(state[name], payload) : updater;
+    const nextState = dotProp.set(state, name, nextValue);
+
+    return nextState;
+  };
+
+  return reducer;
```

Comment:
And another useless stuff :)

Why do you need to invent your own Redux helpers?... Why not just write the code like everybody?
I believe these functions make the code harder to understand without bringing much useful stuff :)

Reply:
We already discussed it. If you provide me the freedom to leave it as it - I'll prefer to keep those helpers :)

Reply:
sure :) Redux docs recommend creating helpers like these in the following article: https://redux.js.org/recipes/structuring-reducers/refactoring-reducer-example/